### ISSUE TITLE
Chrome App v's Extension

### ISSUE THREAD

**Issue by paulhitz** [paulhitz/yet-another-rest-client#22]
Consider a branch for a Chrome Packaged App.

It could be convincingly argued that YARC is more suited to being a Chrome App rather than an extension since it's standalone and doesn't impact the open website. 

The good news is it should be trivial to switch to a Chrome App. Changes should by and large be limited to the manifest file and extension.js.

The icon would no longer appear on the browser task bar.

See also:
https://developer.chrome.com/webstore/choosing
https://developer.chrome.com/webstore/apps_vs_extensions

**Comment by paulhitz** [comment]
Created a new branch with YARC packaged as a working Chrome App.

The vast majority of the tool works well including making requests. 

However there are still some issues regarding accessing some external resources. E.g. Google Fonts, Analytics, paypal etc. Presumably we just need to store them locally.

(Also need to stop Angular from trying to access the browser history state which will fail and pollute the console. Very minor)

Useful doc: https://gist.github.com/maicki/7622137

**Comment by paulhitz** [comment]
YARC can now work as a Chrome Packaged App. See the 'chrome-app' branch.

**Everything appears to work except Google Analytics**. The library (Angulartics) the tool uses for Google Analytics with AngularJS doesn't appear to work with Chrome Apps.

We could role our own using Chrome Platform Analytics (link below) but we wouldn't be able to take advantage of any of the existing Analytics code. Is it worth it? Ideally Angulartics should be used if it can somehow be configured to work with Chrome Packaged Apps.

I've added in some test code that uses Chrome Platform Analytics to track app views and a test event. 

More info: 
https://github.com/GoogleChrome/chrome-platform-analytics/wiki
https://github.com/angulartics/angulartics
https://github.com/angulartics/angulartics-google-analytics

Options:
-Figure out if Angulartics can be configured to work with Chrome Packaged Apps.
-Role our own (more limited) solution.

**Comment by paulhitz** [comment]
Chrome App now available: https://chrome.google.com/webstore/detail/yet-another-rest-client-c/oaelcicjdkdkcioniblhpnanpnhegoad

It was tougher than expected to get 100% of the functionality working. Lots of little issues and a major issue with Google Analytics. I had to use to Chrome Platform Analytics and create my own plugin for Angulartics. I've published it on bower and npm.

Some more testing of the Chrome App is required before it should be made generally available.

Going forward, 2 versions of YARC will be available. The extension and the Chrome App. It will be interesting to see which will be more popular.
